docs/reference/superjson.mdx

```diff
@@ -0,0 +1,294 @@
+---
+slug: /reference/superjson
+---
+
+# Super.json Reference
+
+Super.json is the main Superface configuration file. It contains the configuration about profiles, maps and providers used.
+
+## Anatomy
+
+At the top level, super.json file consists of two sections: [profiles](#profiles) and [providers](#providers). the main purpose profile section is to set up profiles which are to be used, their versions and/or file locations, and their default values. The purpose of provider section is to set up providers that are to be used and their credentials.
+
+```json
+{
+  "profiles": {
+    // ...
+  },
+  "providers": {
+   // ...
+  }
+}
+```
+
+## Profiles
+
+The profiles section consists of a record of all installed profiles. The key is the name of the profile and value is either semantic version string, a file URI or a [ProfileSettings](#profilesettings) object.
+
+```json
+{
+  "profiles": {
+    "myprofile1": "1.0.0",
+    "myprofile2": "file:path/to/profile.supr",
+    "myprofile3": {
+      // ...
+    }
+  }
+}
+```
+
+### ProfileSettings
+
+The ProfileSettings object contains either `file` or `version` property, analogous to the shortcut strings above and following optional properties:
+
+- `priority`: an array of strings containing provider names in desired order for failover
+- `providers`: an array of [ProfileProviderSettings](#profileprovidersettings) objects or URI strings, used for configuring providers for the profile. Not to be confused with top-level Providers record
+- `defaults`: a record of defaults for the usecase, key is usecase name, value is [UsecaseDefaults](#usecasedefaults) object
+
+```json
+{
+  "version": "1.0.0",
+  "priority": ["provider1", "provider2"],
```

Comment:
Provider names **must not** contain numbers, this is invalid.

Reply:
since when? and where is it specified? profile names use DocumentIdentifier afaik (/[a-z][a-z0-9_-]*/)

Reply:
See the schema https://superface.ai/docs/comlink/reference/provider#json-schema
Not sure when this was added (back in August it was discussed as given). The rationale is to prevent typo attacks (eg. `google` vs `goog1e`).

Reply:
and why is it only in this one document (where does the schema even come from?) and not in other documents or in code?

Reply:
I guess it was missing in the spec and wasn't reflected in the authoring guide. I will add an issue to fix it, the authoring guide needs a major revision anyway.